### PR TITLE
#1043 Writer choices set to list

### PR DESCRIPTION
**Pull request by EsipovPA**
Solution proposal for [issue #1043](https://github.com/ros2/rosbag2/issues/1043)

### CODE REVIEW

File: ros2bag/ros2bag/verb/record.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class RecordVerb(VerbExtension):
 
     def add_arguments(self, parser, cli_name):  # noqa: D102
         writer_choices = get_registered_writers()
-        default_writer = 'sqlite3' if 'sqlite3' in writer_choices else writer_choices[0]
+        default_writer = 'sqlite3' if 'sqlite3' in writer_choices else list(writer_choices)[0]
```
Comment (vayerx) [inline ros2bag/ros2bag/verb/record.py:38]:
```suggestion
        default_writer = 'sqlite3' if 'sqlite3' in writer_choices else next(iter(writer_choices))
```

Comment (EsipovPA) [reply ros2bag/ros2bag/verb/record.py]: Looks fine to me. Added here and to reindex.py

**Comment by MichaelOrlov** [comment]
Gist: https://gist.githubusercontent.com/MichaelOrlov/02ac01bb15b80faf24fc6c7d286e55ea/raw/2b4f8b427717dfd86579ee5862a286892cd4b6f8/ros2.repos
BUILD args:  --packages-above-and-dependencies ros2bag rosbag2_tests
TEST args:  --packages-above ros2bag rosbag2_tests
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/10575
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17027/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11587/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17446/)

**Review by MichaelOrlov — APPROVED** [review]
LGTM with "green" CI.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-8-18-2022/27050/1

**Pull request merged**